.github/workflows/build-apk.yml

```diff
@@ -48,13 +49,43 @@ jobs:
       - name: Grant execute permission for gradlew
         run: chmod +x ./gradlew
 
-      # Build the APK
+      # Decode the Keystore from secret
+      - name: Decode Keystore
+        env:
+          ANDROID_KEYSTORE_BASE64: ${{ secrets.ANDROID_KEYSTORE_BASE64 }}
+        run: |
+          mkdir -p ~/.android
+          echo "$ANDROID_KEYSTORE_BASE64" | base64 --decode > ~/.android/debug.keystore
+
+        # Generate SHA1 fingerprint from keystore
+      - name: Generate SHA1
+        run: keytool -list -v -keystore ~/.android/debug.keystore -storepass ${{ secrets.KEYSTORE_PASSWORD }} -keypass ${{ secrets.KEYSTORE_PASSWORD }}
+      # Build the debug APK
       - name: Build APK
         run: ./gradlew assembleDebug
 
+      # Find the latest version of build-tools
+      - name: Find latest build-tools version
+        id: build_tools
+        run: echo "build_tools_version=$(ls ${ANDROID_HOME}/build-tools | sort -V | tail -n 1)" >> $GITHUB_ENV
+
+      # Sign the debug APK using apksigner
+      - name: Sign Debug APK with apksigner
+        run: |
+          ${ANDROID_HOME}/build-tools/${{ env.build_tools_version }}/apksigner sign --ks ~/.android/debug.keystore --ks-key-alias ${{ secrets.KEY_ALIAS }} --ks-pass pass:${{ secrets.KEYSTORE_PASSWORD }} --key-pass pass:${{ secrets.KEYSTORE_PASSWORD }} app/build/outputs/apk/debug/app-debug.apk
+
+      # Verify APK signature
+      - name: Verify APK Signature
+        run: |
+          ${ANDROID_HOME}/build-tools/${{ env.build_tools_version }}/apksigner verify app/build/outputs/apk/debug/app-debug.apk
+
+      # Verify APK Signature using SHA1
```

Comment:
This doesn't verify APK signature per se, instead it prints them, so allows for visual verification of SHA1. Specify in comment and step name to make it clearer maybe

Reply:
yes, I will change that too